src/porepy/numerics/interface_laws/elliptic_interface_laws.py

```diff
@@ -723,4 +725,260 @@ def assemble_matrix_rhs(
         return matrix, rhs
 
 
-# ------------------------------------------------------------------------------
+class RobinContactBiotPressure(RobinContact):
+    """
+    This condition adds the fluid pressure contribution to the Robin contact condition.
+    The Robin condition says:
+    MW * lambda + RW * [u] = robin_rhs,
+    where MW (mortar_weight) and RW (robin_weight) are two matrices, and
+    [u] = u_slave - u_master is the displacement jump from the slave to the master.
+    In Biot the displacement on the  contact boundary (u_slave and u_master) will be a
+    linear function of cell center displacement (u), mortar stress (lambda) and cell
+    centere fluid pressure (p):
+        A * u + B * p + C * lam = u_slave/u_master
+    This class adds the contribution B.
+    
+    To enforce the full continuity this interface law must be used in combination with
+    the RobinContact conditions which adds the contributions A and C
+    """
+
+    def discretize(self, g_h, g_l, data_h, data_l, data_edge):
+        """ Discretize the robin weight (RW)
+        
+        Parameters:
+            g_h: Grid of the master domanin.
+            g_l: Grid of the slave domain.
+            data_h: Data dictionary for the master domain.
+            data_l: Data dictionary for the slave domain.
+            data_edge: Data dictionary for the edge between the domains.
+
+        """
+        matrix_dictionary_edge = data_edge[pp.DISCRETIZATION_MATRICES][self.keyword]
+        parameter_dictionary_edge = data_edge[pp.PARAMETERS][self.keyword]
+
+        robin_weight = sps.block_diag(parameter_dictionary_edge["robin_weight"])
+        matrix_dictionary_edge["robin_weight"] = robin_weight
+
+    def assemble_matrix_rhs(
+        self, g_master, g_slave, data_master, data_slave, data_edge, matrix
+    ):
+        """ Assemble the dicretization of the interface law, and its impact on
+        the neighboring domains.
+        Parameters:
+        ----------
+            g_master: Grid on one neighboring subdomain.
+            g_slave: Grid on the other neighboring subdomain.
+            data_master: Data dictionary for the master suddomain
+            data_slave: Data dictionary for the slave subdomain.
+            data_edge: Data dictionary for the edge between the subdomains
+            matrix_master: original discretization for the master subdomain
+            matrix_slave: original discretization for the slave subdomain
+
+        """
+        matrix_dictionary_edge = data_edge[pp.DISCRETIZATION_MATRICES][self.keyword]
+
+        self.discretize(g_master, g_slave, data_master, data_slave, data_edge)
+
+        if not g_master.dim == g_slave.dim:
```

Comment:
This means that for a fracture problem, g_master == g_slave? Not necessarily, but in practice.

Reply:
No, g_master can be a different domain than g_slave, both in theory and practice.